### PR TITLE
Create dndTransferCases.html

### PR DESCRIPTION
**Pull request by ayg**
Created from #323 because I broke it by mistake.  :(  I think I copied the commit correctly, except that I couldn't get GitHub to tell me the exact date.

@cc17

### CODE REVIEW

**Comment by ayg** [comment]
Note: original PR was from 2013.

File: html/editing/dnd/the-datatransfer-interface/dndTransferCases.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    <script src='/resources/testharness.js'></script>
+    <script src='/resources/testharnessreport.js'></script>
+    <style>
+      #drop {
+        border: 2px solid black;
+        width: 100px;
+        height: 100px;
+        padding: 20px;
+      }
+      #select {
```
Comment (ayg) [inline html/editing/dnd/the-datatransfer-interface/dndTransferCases.html]: You probably mean #drag here?  The blue text isn't blue.

File: html/editing/dnd/the-datatransfer-interface/dndTransferCases.html [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  </head>
+  
+  <body>
+    <div>Select and drag the blue text to rectangular box.</div>
+    <div id='drag' draggable>blue text</div>
+    <div id='drop' dropzone='copy string:text/plain'></div>
+    <div id='log'> </div>
+
+    <script>
+      var drop;
```
Comment (ayg) [inline html/editing/dnd/the-datatransfer-interface/dndTransferCases.html]: You probably mean "var drag"?  Using drag without declaring seems to work, but AFAIK it's not considered correct JS practice (creates window.drag?).

File: html/editing/dnd/the-datatransfer-interface/dndTransferCases.html [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      setup(function() {
+          drag = document.querySelector('#drag');
+      }, {explicit_done: true, explicit_timeout: true});
+      
+      on_event(drag, 'dragstart', function(event) {
+        var dataTransfer = event.dataTransfer;
```
Comment (ayg) [inline html/editing/dnd/the-datatransfer-interface/dndTransferCases.html]: Why do you need this variable if you only refer to it in one place?  Just reference event.dataTransfer directly.

File: html/editing/dnd/the-datatransfer-interface/dndTransferCases.html [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        test(function() {
+          assert_equals(dataTransfer.effectAllowed, '',  'effectAllowed show be empty');
```
Comment (ayg) [inline html/editing/dnd/the-datatransfer-interface/dndTransferCases.html]: I think you mean "should" instead of "show", here and on the next line.

**Comment by ayg** [comment]
This looks like a manual test, so I think it has to be marked somehow.  @jgraham, what's the procedure?  It has to be called .manual.html or something?

File: html/editing/dnd/the-datatransfer-interface/dndTransferCases.html (hunk 4 above)
Comment (ayg) [inline html/editing/dnd/the-datatransfer-interface/dndTransferCases.html]: AFAICT, in the current spec, it should be "uninitialized", not empty:
> When a drag data store is created, it must be initialised such that . . . its drag data store allowed effects state is the string "uninitialized".
https://html.spec.whatwg.org/#drag-data-store-allowed-effects-state

And that's what Edge/Firefox/Chrome do

**Comment by ayg** [comment]
Other than that, looks good.

**Comment by ayg** [comment]
Does someone else want to review this with my fixes, then merge *without* squashing (so that attribution is correctly maintained)?

**Comment by zcorpan** [comment]
> "uninitialized" when what is being dragged is a selection

https://html.spec.whatwg.org/multipage/interaction.html#fire-a-dnd-event

Test passes in Chromium/WebKit/Gecko.

LGTM, I can merge.

**Comment by zcorpan** [comment]
Oh you beat me to it, very well :-)

**Pull request merged**